### DISCUSSION TITLE
Can Prestashop handle 1.5 million products and 30,000 categories with a category depth of 6?

### DISCUSSION THREAD

**Discussion by francoiscpaphil** [PrestaShop/PrestaShop#32694] [Q&A]
Hello guys, 

I am considering using Prestashop for my e-commerce project, and I would like to know if it can handle my specific use case. I have around 1.5 million products distributed across nearly 30,000 different categories. The depth of the category structure is 5 levels deep. 

My questions are: 

Can Strapi efficiently handle this large number of products and categories without any significant performance issues? 

Are there any known limitations or issues when dealing with such a large and complex category structure in Prestashop ? 

Are there any recommendations or best practices for managing this scale of products and categories with Prestashop  ? 

I would greatly appreciate your insights and advice on whether Prestashop  is suitable for this use case. Thank you for your time and assistance. 

Best regards, Francois

**Comment by matks** [comment]
> Can Strapi efficiently handle this large number of products and categories without any significant performance issues?

No idea for Strapi 😆

**Comment by matks** [comment]
Now that I made my joke 😆 time for a better answer.

I'm afraid I might not be the best person to answer because I'm no expert in how to scale PrestaShop. I know the software well, I know how to maintain it and make it evolve but I do not have a production environment with PrestaShop. But in the PrestaShop community there are lot of contributors who run agencies and they do run PrestaShop in production 😄 dealing with scaling, performance, and large catalogs. I hope some of them answer.

The question you ask is too broad. To get an accurate answer you would need to be a lot more specific:
What is "efficiently" ?
What infrastructure are you going to run PrestaShop on?
What is "complex" ?

I think unfortunately the only answer is "it depends", and it depends on _so many things_. Traffic, CPU, memory, database configuration, infrastructure, average number of products in the cart, etc, etc.

I can give you this link https://build.prestashop-project.org/news/2020/performance-and-prestashop/ it will tell you a few things and raise even more questions.

I can tell you companies like Decathlon or Carrefour have been using PrestaShop just fine, and I guess they have some dozen (maybe) hundred of thousand products. So it's definitely possible. But out-of-the-box? I doubt it. I think you'd need some fine-tuning of the infrastructure behind. Remember that PrestaShop is just the software, just the application layer. Performance and scaling happen when both the software and the environment it runs in are carefully built 😉 .

**Comment by kpodemski** [comment]
The number of products PrestaShop can handle is not the ultimate factor in whether the software suits you, but realistically speaking, it would be a challenge.

You would, for sure, encounter several issues when working with such an extensive catalog:
- you would need a customized product management page to handle the category tree this big
- you would need to be extra careful when presenting your offer in the front office, meaning one module that would read too many products at once could break your store
- you would need a separate tool to perform bulk editing (maybe Store Commander).

And there would be more. There are also several known PrestaShop limitations:
- poor performance for products with many combinations
- a few places in the core that are slow when you have a big catalog, for example, PrestaShop struggles with slow product deletion

You may be surprised to see my answer, an answer from somebody working on PrestaShop, but I don't want to lie to you and write that everything's gonna be ok because it would be challenging but not impossible! :)

**Reply by francoiscpaphil** [reply]
thanks, pretty helpful ! We contemplate other solutions (ie Strapi), but have to see and discuss what is most relevant! Merci (thanks) !
